core/domain/classroom_config_services.py

```diff
@@ -327,4 +338,43 @@ def delete_classroom(classroom_id: str) -> None:
     Args:
         classroom_id: str. ID of the classroom which is to be deleted.
     """
+    classrooms = get_all_classrooms()
+
+    for classroom in classrooms:
+        if classroom.classroom_id == classroom_id:
+            index_to_delete = classroom.index
+            break
+
+    for classroom in classrooms:
+        if classroom.index > index_to_delete:
+            classroom.index -= 1
+            update_classroom(classroom)
+
     classroom_models.ClassroomModel.get(classroom_id).delete()
+
+
+@transaction_services.run_in_transaction_wrapper
+def update_classroom_id_to_index_mappings(
+    classroom_index_mappings: List[
+        classroom_config_domain.ClassroomIdToIndexDict
+    ]
+) -> None:
+    """Updates the index of multiple classrooms.
+
+    Args:
+        classroom_index_mappings: List[ClassroomIdToIndex]. The domain
+            objects for the given mapping.
+
+    Raises:
+        Exception. No mapping found for the given classroom ID.
+    """
+
+    for classroom_index_mapping in classroom_index_mappings:
+        classroom_id = classroom_index_mapping['classroom_id']
+        classroom_name = classroom_index_mapping['classroom_name']
+        classroom_index = int(classroom_index_mapping['classroom_index'])
+
+        classroom = get_classroom_by_id(classroom_id)
+        classroom.name = classroom_name
```

Comment:
This is strange, this handler and this function shouldn't allow updating the name, or am I wrong?

Reply:
Yes it should not update it. removed this.